storage/ndb/rest-server/rest-api-server/internal/handlers/handler.go

```diff
@@ -17,16 +17,29 @@
 
 package handlers
 
-import "net/http"
+import (
+	"errors"
+	"net/http"
+	"runtime/debug"
+
+	"hopsworks.ai/rdrs/internal/log"
+)
 
 type Handler interface {
 	Validate(request interface{}) error
 	Authenticate(apiKey *string, request interface{}) error
 	Execute(request interface{}, response interface{}) (int, error)
 }
 
-func Handle(h Handler, apiKey *string, request interface{}, response interface{}) (int, error) {
-
+func Handle(h Handler, apiKey *string, request interface{}, response interface{}) (statusCode int, err error) {
+	defer func() {
+		if r := recover(); r != nil {
+			statusCode = http.StatusInternalServerError
+			err = errors.New(`{"message": "Internal server error."}`)
+			log.Error(r.(error).Error())
```

Comment:
is this thread safe?

Reply:
You mean the log library? Should be, yes.

Reply:
You can ignore the comment in the GO code regarding the thread-safe logger. What I meant there was just to use a Go logger instance/object, like zap.logger.